web/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -9,20 +10,43 @@ import { AuthService } from 'src/app/services/auth.service';
 })
 export class DashboardComponent implements OnInit {
 
+  public user: User;
+  public userSocial: UserSocial;
+  public isResponseLoading: boolean;
+
   constructor(
-    private authService: AuthService,
-    private router: Router,
+    private userService: UserService,
   ) { }
 
-  ngOnInit() { }
-
-  logout() {
-    this.authService.signOutUser()
-      .subscribe(() => this.navigateTologin());
+  ngOnInit() {
+    this.isResponseLoading = true;
+    this.userService.getUser()
+      .subscribe((user) => {
+        if (!user) {
+          console.error('error in fetching user'); // TODO: Handle null user scenario in dashboard UI
+          return;
+        }
+        this.user = { ...user };
```

Comment:
Is this needed?

Reply:
yes @eddiejaoude . As we are cloning from argument. I thought of creating new memory reference before storing it in the class variable.

Reply:
Ok